src/subcommands/execution_results_summary/summary.rs

```diff
@@ -0,0 +1,122 @@
+use std::collections::BTreeMap;
+
+use casper_types::{bytesrepr::ToBytes, ExecutionResult};
+use serde::{Deserialize, Serialize};
+
+use super::Error;
+
+#[cfg(not(test))]
+pub(crate) const CHUNK_SIZE_BYTES: usize = 8 * 1024 * 1024;
+#[cfg(test)]
+pub(crate) const CHUNK_SIZE_BYTES: usize = 20;
+const LAST_ELEM_INDEX_IN_CHUNK: usize = CHUNK_SIZE_BYTES - 1;
+
+#[inline]
+pub(crate) fn chunk_count_after_partition(element_count: usize) -> usize {
+    (element_count + LAST_ELEM_INDEX_IN_CHUNK) / CHUNK_SIZE_BYTES
+}
+
+fn summarize_map(map: &BTreeMap<usize, usize>, elem_count: usize) -> CollectionStatistics {
+    let median_pos = elem_count / 2;
+    let mut sum = 0usize;
+    let mut current_idx = 0usize;
+    let mut median = 0usize;
+    let mut max = 0usize;
+    for (key, count) in map.iter() {
+        if current_idx <= median_pos && current_idx + count > median_pos {
+            median = *key;
+        }
+        sum += *key * *count;
+
+        current_idx += count;
+        if current_idx == elem_count {
+            max = *key;
+        }
+    }
+    let average = if elem_count > 0 {
+        sum as f64 / elem_count as f64
+    } else {
+        0.0
+    };
+
+    CollectionStatistics {
+        average,
+        median,
+        max,
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct ExecutionResultsStats {
+    pub execution_results_size: BTreeMap<usize, usize>,
+    pub chunk_count: BTreeMap<usize, usize>,
+    pub execution_results_insert_counter: usize,
+}
+
+impl ExecutionResultsStats {
+    pub fn feed(&mut self, execution_results: Vec<ExecutionResult>) -> Result<(), Error> {
+        // Calculate the length of the bincode serialized execution
+        // results.
+        let bincode_encoded_execution_results_length =
```

Comment:
Nit:

```suggestion
        let bincode_encoded_execution_results_size =
```

OTOH: I'm not sure, since we're already being inconsistent (`serialized_size()` vs `serialized_length()`), so up to you.

Reply:
Will do the suggestion.